### PR TITLE
Cisco logical channels

### PR DESCRIPTION
**Pull request by karthikeya-remilla**
Changes for zr_logical_channels_test:
1. operationalMode flag
2. cisco numbering deviations for eth and otn channels
3. eth and otn channels' index testcases based on deviations

Note: ciscoOpticalChannelFormat := strings.ReplaceAll(opticalChannel, "/", "_") // Ex: OpticalChannel0_0_0_18

This is temporary line under the deviation till Cisco supports OpticalChannel in standard OpticalChannel0/0/0/18 format. Internal bug raised and in progress.

### CODE REVIEW

File: feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go
Comment (ejbrever) [inline feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go:151]: can you define a struct type near the top:

type testcase struct {
	desc string
	got  any
	want any
}

Then around line 105, you can have:

tcs := []testcase {
	<testcases>
}

Then here, this just changes to:

assignmentIndexTestCase := testcase{
	desc: "Assignment: Index",
	got: ec.GetAssignment(0).GetIndex(),
	want: uint32(0),
})
if deviations.EthChannelAssignmentCiscoNumbering(dut) {
	assignmentIndexTestCase.want = uint32(1)
}
tcs = append(tcs, assignmentIndexTestCase)

File: feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go
Comment (ejbrever) [inline feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go:229]: please consider previous comment to avoid lots of redeclaring this struct definition.

File: feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go
Comment (karthikeya-remilla) [reply feature/platform/transceiver/tests/zr_logical_channels_test/zr_logical_channels_test.go:151]: Sure, will do!

**Comment by karthikeya-remilla** [comment]
@ejbrever could you please take a look at the improved code now?

**Review by ejbrever — APPROVED** [review]
Thanks for the updates.

**Comment by karthikeya-remilla** [comment]
Waiting for approval from @ahsaanyousaf

**Review by ahsaanyousaf — APPROVED** [review]

**Review by ahsaanyousaf — APPROVED** [review]

**Review by amrindrr — DISMISSED** [review]

**Comment by dplore** [comment]
@amrindrr please run and verify this and if good, you can merge it

**Comment by ahsaanyousaf** [comment]
Please resolve the conflicts so this can be merged.

**Pull request merged**
